OpenRA.Mods.Common/Traits/DamagedByTerrain.cs

```diff
@@ -59,7 +59,7 @@ public DamagedByTerrain(Actor self, DamagedByTerrainInfo info)
 
 		void INotifyAddedToWorld.AddedToWorld(Actor self)
```

Comment:
Shouldn't this be changed to `TraitEnabled` to avoid issues with not initialising `damageThreshold`?

Reply:
Not sure why it is in AddedToWorld at the first place. I would expect it to be on Created before and if we want it to reset to Threshold every time trait is enabled, makes sense for it to be on TraitEnabled. Could also change the trait to PausibleConditional so pausing don't recheck the StartOnThreshold, but disabling does.

Reply:
I don't know what the usecase for making it pausable would be, so am indifferent on that. Thinking about it, `Created` might not work for the same reason as just `TraitEnabled`: What if the actor is created, but not added to the world directly? We probably need a combination of `Created`/`AddedToWorld` and `TraitEnabled` to cover the cases of a) actor created but only later added to the world, b) actor created and added to the world, but trait enabled later. (?)

I suppose we could get away with keeping the calculations in `AddedToWorld` but only deal the initial damage when the trait is not disabled.

Reply:
Can an actor that isn't in world not get damaged? Why would you want to delay this until it is added to world? Currently with AddedToWorld it would deal the damage every time a unit is removed and added back to world, i don't think this would be intended.

Reply:
It doesn't really occupy space when not in the world, so damaging on foundation doesn't make sense.

Reply:
Yeah, you are right. I forgot which trait we were dealing with. Still no idea what exactly to do tho.

Reply:
I'll file a follow-up PR, as I noticed it's a pre-existing bug and it's kinda unfair to block this PR (and the milestone) over it.